tests/test_drsclient.py

```diff
@@ -0,0 +1,281 @@
+import json
+import pytest
+import uuid
+import requests
+from drsclient import __version__
+from drsclient.client import DrsClient
+import asyncio
+import respx
+
+from cdisutilstest.code.indexd_fixture import (
+    create_random_index,
+    create_random_index_version,
+)
+
+
+def test_version():
+    assert __version__ == "0.1.0"
+
+
+def get_index_doc():
+    doc = {
+        "form": "object",
+        "size": 123,
+        "urls": ["s3://endpointurl/bucket/key"],
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+    }
+    return doc
+
+
+def create_index_record(index_client):
+    i_data = get_index_doc()
+    res1 = index_client.create(
+        hashes=i_data["hashes"], size=i_data["size"], urls=i_data["urls"]
+    )
+    rec1 = res1._doc
+    return rec1
+
+
+def test_get_latest_version(index_client):
+    """
+    Args:
+        index_client (indexclient.client.IndexClient): IndexClient Pytest Fixture
+    """
+    doc = create_random_index(index_client)
+    latest = index_client.get_latest_version(doc.did)
+    assert latest.did == doc.did
+    assert latest.file_name == doc.file_name
+    assert latest.hashes == doc.hashes
+
+
+def test_check_status(index_client, drs_client):
+    res = drs_client.check_status()
+    print(res.status_code)
+    assert res.status_code == 200
+
+
+def test_post_index(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res2 = drs_client.get(guid=did)
+    assert res2.status_code == 200
+
+
+def test_create_bundle(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res1 = drs_client.create(bundles=[did])
+    rec1 = res1.json()
+    contents = json.loads(rec1["contents"])
+    assert res1.status_code == 200
+    assert len(contents) == 1
+    assert contents[0]["id"] == did
+
+
+def test_async_create_bundle(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res1 = asyncio.run(drs_client.async_create(bundles=[did]))
+    rec1 = res1.json()
+    assert res1.status_code == 200
+    contents = json.loads(rec1["contents"])
+    assert len(contents) == 1
+    assert contents[0]["id"] == did
+
+
+def test_get_drs_record(index_client, drs_client):
+    rec = create_index_record(index_client)
+    did = rec["did"]
+    res2 = drs_client.get(guid=did)
+    assert res2.status_code == 200
+
+
+def test_get_bundle(index_client, drs_client):
```

Comment:
for the tests i think you could parametrize async_client=True/False to avoid duplication